src/SourceGenerators/Uno.UI.SourceGenerators/XamlGenerator/XamlFileGenerator.cs

```diff
@@ -3276,7 +3276,23 @@ private void BuildExtendedProperties(IIndentedStringBuilder outerwriter, XamlObj
 
 								var propertyType = GetPropertyTypeByOwnerSymbol(ownerType, member.Member.Name);
 
-								if (IsExactlyCollectionOrListType(propertyType))
+								if (member.Objects.Count == 1 && SymbolEqualityComparer.Default.Equals(propertyType, Generation.ResourceDictionarySymbol.Value))
```

Comment:
I guess the right condition might be that propertyType equals the type of `member.Objects[0]`. I'll need to do more testing.

Reply:
Also this code seems to be a duplicate of the `else` block below. We need to figure out the right conditions and their order.